README.md

```diff
@@ -1,41 +1,138 @@
+| ------------- | ------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| loadReference | `"false"`     | Does not do anything, however, is supported per the [Okapi specification](https://github.com/folio-org/okapi/blob/master/doc/guide.md)                                                                                                                                                                                        |
+| loadSample    | `"false"`     | If `"true"`, a series of sample calendars will be added representing examples of exceptions and openings across multiple calendars and service points. This will use the same service points loaded by [mod-inventory-storage](https://github.com/folio-org/mod-inventory-storage/tree/master/reference-data/service-points). |
 
-See the built `target/ModuleDescriptor.json` for the interfaces that this module
-requires and provides, the permissions, and the additional module metadata.
+### Environment variables
 
-### API documentation
+See `.env.sample` for example values.
+
+| Name            | Default Value              | Description                                                                           |
+| --------------- | -------------------------- | ------------------------------------------------------------------------------------- |
+| JAVA_OPTIONS    | `XX:MaxRAMPercentage=66.0` | Sets `java` command-line options. Default is to set the maximum ram percentage to 66% |
+| DB_HOST         | `localhost`                | Postgres hostname                                                                     |
+| DB_PORT         | `5432`                     | Postgres port                                                                         |
+| DB_USERNAME     | `folio_admin`              | Postgres username                                                                     |
+| DB_PASSWORD     | `folio_admin`              | Postgres password                                                                     |
+| DB_DATABASE     | `okapi_modules`            | Postgres database name                                                                |
+| DB_QUERYTIMEOUT | `60000`                    | Postgres query time out. Default is 6s                                                |
+| DB_CHARSET      | `UTF-8`                    | Postgres charset                                                                      |
+| DB_MAXPOOLSIZE  | `5`                        | Postgres max pool size                                                                |
+
+#### Integration Tests
+
+Integration tests have special environment variables that control whether or not API requests are
+routed through a proxy:
```

Comment:
What kind of circumstances would folks want to run those tests using a proxy?

Reply:
When debugging tests, I found it quite useful -- at the least, it was a lot easier than adding a bunch of print statements around each restassured call to make sure everything is working as expected.  It's disabled by default, but I figured why not leave it in, perhaps it could be useful to another developer in the future.

Reply:
bq. When debugging tests, I found it quite useful -- at the least, it was a lot easier than adding a bunch of print statements around each restassured call to make sure everything is working as expected.

Thank you for that context. Maybe it could be worth explaining that use case alongside these instructions.

How does the proxy allow you to get that information better?

> It's disabled by default, but I figured why not leave it in, perhaps it could be useful to another developer in the future.

That's up to you. I think that depends upon the maintenance cost.

Reply:
I'll add some context to the README, that's a great idea.

Here's an example, for a test that checks pagination works as expected.

Maven/JUnit by default will log this:
![image](https://user-images.githubusercontent.com/8005215/186042907-fb86587c-79b6-462c-a9b3-8d464e8274fd.png)

It won't even denote when a test method itself ends or begins, only by class (unless there's a failure), which can make searching difficult.  Additionally, although it logs the endpoints and query parameters, anything contained in the body is not logged by default and would require manually going into the test case, copying whatever was being sent to the endpoint, and logging it -- something that can become pretty tedious pretty fast.

With the proxy, it's a little faster to drill down into the requests themselves.  It provides a quick and easy GUI to inspect requests and responses, which has saved me a couple headaches, especially when trying to debug response codes (RA likes to print out that the response code failed and no other error information, the proxy can show the full response at a glance).
![image](https://user-images.githubusercontent.com/8005215/186042827-d78addd4-e89d-469a-b7c6-a010b224094b.png)

The maintenance cost is pretty slim.  The main part of this code is in `BaseApiTest::initialize`, and really just pulls out the environment variables and puts them in `RestAssured.proxy`, then RA handles the rest.

Reply:
Thank you for the detailed explanation.

> Here's an example, for a test that checks pagination works as expected.

Is that logging from the production code or tests?
 
 > Additionally, although it logs the endpoints and query parameters, anything contained in the body is not logged by default and would require manually going into the test case, copying whatever was being sent to the endpoint, and logging it -- something that can become pretty tedious pretty fast.

What is it, RestAssured or the production code or something else?

Reply:
Tests only, there's no use of RA/proxy in the production code.  Any production proxying would have to be done from whatever application is making the requests.

> I thought RestAssured could be configured to log everything about a request or response.

Didn't know that was a thing until just now, looks like it can by adding .log().all() to certain requests

Reply:
> Tests only, there's no use of RA/proxy in the production code. Any production proxying would have to be done from whatever application is making the requests.

I meant the source of the logging rather than the proxying. Apologies for the ambiguity.

> Didn't know that was a thing until just now, looks like it can by adding .log().all() to certain requests

Yes, I believe so. I think it can also be done via specifications that can be reused across RA requests / responses.

Reply:
Source of the logging is somewhere in folio-spring-base